dhall/src/Dhall/Eval.hs

```diff
@@ -527,9 +527,13 @@ eval !env t0 =
                                 -- following issue:
                                 --
                                 -- https://github.com/ghcjs/ghcjs/issues/782
-                                let go !acc 0 = acc
-                                    go  acc m = go (vApp succ acc) (m - 1)
-                                in  go zero (fromIntegral n' :: Integer)
+                                go zero (fromIntegral n' :: Integer) where
+                                  go !acc 0 = acc
+                                  go (VNaturalLit x) m =
```

Comment:
Here, we detect the shortcut only when the accumulator is a `Natural` literal. However, we would like to detect the shortcut in all cases. I could not find any way of comparing `Val` values. What kind of code would need to be written here?

Reply:
`Dhall.Eval.conv` is what you want